assembly/src/lib.rs

```diff
@@ -49,16 +49,23 @@ type ModuleMap = BTreeMap<String, ProcMap>;
 pub struct Assembler {
     stdlib: StdLibrary,
     parsed_modules: ModuleMap,
+    in_debug_mode: bool,
 }
 
 impl Assembler {
     // CONSTRUCTOR
     // --------------------------------------------------------------------------------------------
     /// Returns a new instance of [Assembler] instantiated with empty module map.
     pub fn new() -> Self {
+        Self::with_debug(false)
+    }
+
+    /// Returns a new instance of [Assembler] instantiated with empty module map in debug mode.
+    pub fn with_debug(in_debug_mode: bool) -> Self {
         Self {
             stdlib: StdLibrary::default(),
             parsed_modules: BTreeMap::new(),
+            in_debug_mode,
         }
     }
```

Comment:
I don't think `Assembler` is instantiated in too many places. Maybe in this case we don't need `with_debug()` and just pass the flag into the `new()` constructor? We could also implement `Default` trait for the `Assembler` which would instantiate not in debug mode.

Reply:
Removed `with_debug()`

Reply:
Removed `with_debug()` and used default for non-debug mode.